lib/logstash/filters/jdbc_static.rb

```diff
@@ -0,0 +1,217 @@
+  #   }
+  # ]
+  # This is optional. You can provide a pre-populated local database server then no initial loaders are needed.
+  config :loaders, :required => false, :default => [], :validate => [LogStash::Filters::Jdbc::Loader]
+
+  # Define an array of Database Objects to create when the plugin first starts.
+  # These will usually be the definitions to setup the local in-memory tables.
+  # For example:
+  # local_db_objects => [
+  #   {name => "servers", preserve_existing => true, index_columns => ["ip"], columns => [["id", "INTEGER"], ["ip", "varchar(64)"], ["name", "varchar(64)"], ["location", "varchar(64)"]]},
+  # ]
+  # NOTE: Important! use `preserve_existing => true` to keep a table created and filled in a previous Logstash session. It will default to false and is unneeded if the database is not persistent.
+  # NOTE: Important! Tables created here must have the same names as those used in the `loaders` and
+  # `local_lookups` configuration options
+  config :local_db_objects, :required => false, :default => [], :validate => [LogStash::Filters::Jdbc::DbObject]
+
+  # Define the list (Array) of enhancement local_lookups to be applied to an event
+  # Each entry is a hash of the query string, the target field and value and a
+  # parameter hash. Target is overwritten if existing. Target is optional,
+  # if omitted the lookup results will be written to the root of the event like this:
+  # event.set(<column name (or alias)>, <column value>)
+  # Use parameters to have this plugin put values from the event into the query.
+  # The parameter maps the symbol used in the query string to the field name in the event.
+  # NOTE: when using a query string that includes the LIKE keyword make sure that
+  # you provide a Logstash Event sprintf pattern with added wildcards.
+  # For example:
+  # local_lookups => [
+  #   {
+  #     "query" => "select * from country WHERE code = :code",
+  #     "parameters" => {"code" => "country_code"}
+  #     "target" => "country_details"
+  #   },
+  #   {
+  #     "query" => "select ip, name from servers WHERE ip LIKE :ip",
+  #     "parameters" => {"ip" => "%{[response][ip]}%"}
+  #     "target" => "servers"
+  #   }
+  # ]
+  config :local_lookups, :required => true, :validate => [LogStash::Filters::Jdbc::Lookup]
+
+  # Schedule of when to periodically run loaders, in Cron format
+  # for example: "* * * * *" (execute query every minute, on the minute)
+  #
+  # There is no schedule by default. If no schedule is given, then the loaders are run
+  # exactly once.
+  config :loader_schedule, :validate => [LogStash::Filters::Jdbc::LoaderSchedule]
+
+  # Append values to the `tags` field if sql error occured
+  config :tag_on_failure, :validate => :array, :default => ["_jdbcstreamingfailure"]
+
+  # Append values to the `tags` field if no record was found and default values were used
+  config :tag_on_default_use, :validate => :array, :default => ["_jdbcstreamingdefaultsused"]
```

Comment:
Is `_jdbcstreamingdefaultsused` correct?

Reply:
Yes I think so. If a default is used then it means that a reference record could not be found - that should trigger the user to investigate why.

Reply:
I was wondering whether _streaming_ was the right name rather than _static_

Reply:
My bad, copy pasta from jdbc-streaming :-(